x-pack/plugins/connectors_networking/server/docs/poc.md

```diff
@@ -0,0 +1,345 @@
+customers to update connectors, since for any field they want to access, they'll
+need to re-enter the secrets.  This would be even more painful with PEM-encoded
+certs, especially if a chain was involved.  Feels like we can return these 
+in at least some APIs, to allow easier editing of the CNO's over time, if the
+`tls.ca` field isn't changing but other things are.  Fair to say sometimes
+you're trying all kinds of TLS options when trying to get a connection to work,
+and having to re-enter the certs in those situations would get real old real
+fast.
+
+In the future, we may need to store REAL secrets in here - client certs and
+keys, specifically.  Those would absolutely need to be in an ESO, and we'd
+have to think about cases where we might want them returned in HTTP responses.
+
+## remaining work items till this is deliverable
+
+- store CNO's in encrypted saved objects
+- complete work for customizing agents 
+- feature controls / privileges
+- support for smtp via nodemailer
+- support for slack using it's SDK
+- management ui for CNO's - CRUD support
+- enhance existing connectors/actions to show applicable CNO, link for more info
+- docs
+- tests
+- feature flag?
+
+## future work
+
+- I think moving all the axios-related code in the actions plugin into this
+  new plugin makes sense.  It will allow us to largely hide axios as an
+  implementation detail, but more importantly cut down on the amount of
+  cross-talk required between actions and this new plugin.  Probably means
+  also moving networking config properties in actions to the new plugin,
+  so that will require some deprecation.
+
+## trying it out
+
+A bash script is provided in this file:
+
+- `x-pack/plugins/connectors_networking/server/routes/test.sh`
+
+The script creates a webhook that points back into Kibana, posting to a 
+bogus link.  Since running Kibana locally with security on implies you are
+running Kibana with a self-signed certificate, we can use this non-functional
+url to compare running with `rejectAuthenticated` as `true` and `false` to 
+make sure the property is being applied.  When `true` the webhook should fail
+with a TLS self-signed cert error, and when `false` the webhook should fail
+with a plain old 404 error, which implies it made it through the TLS auth.
+
+It also runs a test using the ca that is used to sign the cert that Kibana
+runs with in dev mode.  This ensures that you can actually set a per-server
+ca and have it used on the request.
+
+Enable debug logging via the following in your `kibana.dev.yml`:
+
+```
+logging:
+  loggers:
+    - name: plugins.actions
+      level: debug
+      appenders: [console]
+    - name: plugins.connectorsNetworking
+      level: debug
+      appenders: [console]
+```
+
+## open questions
+
+- Names!  connectorsNetworking?  erghh.  should we use actions instead?
+
+- Do we need to make the saved objects hidden?  I don't think so, but worth
```

Comment:
This largely depends on your security requirements. It sounds like you don't need anything fancy like the producer/consumer model you have for alerts/actions, so I suspect you're right about not needing to make these hidden.